### PR TITLE
converter: apply whitelist filtering on supported types

### PR DESCRIPTION
**Pull request by mfn**
Only in effect on all-lowercase types. This is to filter out stuff like
`mixed`, `void`, `$this`, etc.

Currently return type hints like `mixed`, `void` or `$this`, which are quite common, get converted literally into PHP.

`mixed` and `void` are expected to be classes, `$this` isn't an allowed syntax.

The idea of this PR is that, in case the whole type hint is in lowercase, it must be one of the supported type declarations as listed at http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration ; otherwise the type hint gets just ignroed.

This automatically filters out a lot of unsupported stuff.

An edge case is when referring to all-lowercase classes. This PR wouldn't support them.

If this is acceptable, I can provide tests too.

### CODE REVIEW

**Comment by dunglas** [comment]
👍! Can you provide a test?

Btw, can you add support for common aliases like `integer` for `int`? Here is a list of common aliases: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php#L324-L346

**Comment by mfn** [comment]
- rebased against master
- added normalization
- added a test

File: src/Converter.php [hunk 1]
```diff
@@ -517,4 +561,38 @@ private function endsInNewLine(string $output): string
...
+                return 'bool';
+
+            // real is not part of the PHPDoc standard, so we ignore it
+            case 'double':
+                return 'float';
+
+            case 'callback':
+                return 'callable';
+
+            case 'void':
```
Comment (dunglas) [inline src/Converter.php]: This one can be removed.

Comment (mfn) [inline src/Converter.php]: Thx, force-pushed an update!

**Comment by dunglas** [comment]
Thank you!
